### PR TITLE
Some changes based on static code analysis

### PR DESCRIPTION
**Pull request by DKenefake**
I made some changes based on the recent integration to codacy here. This is mostly focusing on the mutable default arg, prng, and whitespace changes. I will put some comments here at locations to make sure everything is aligned.

### CODE REVIEW

File: src/energiapy/graph/graph.py [hunk 1]
```diff
@@ -25,17 +24,18 @@ def distance(u_x: float, u_y: float, v_x: float, v_y: float, method: str = 'eucl
     Returns:
         float: _description_
     """
-    if 'rectilinear':
-        distance = abs(u_x-v_x) + abs(u_y-v_y)
+    if method == 'rectilinear':
```
Comment (DKenefake) [inline src/energiapy/graph/graph.py:27]: wouldn't return an initialized value; if an unknown distance method is used throws a value error.

File: src/energiapy/graph/graph.py [hunk 2]
```diff
@@ -25,17 +24,18 @@ def distance(u_x: float, u_y: float, v_x: float, v_y: float, method: str = 'eucl
...
 
-    elif 'chebyshev':
-        distance = max(abs(u_x-v_x) + abs(u_y-v_y))
-    elif 'euclidean':
-        distance = math.sqrt((u_x-v_x)**2 + (u_y-v_y)**2)
-    return distance
 
-
-def make_graph(n_sources: int, n_sinks: int, n_facilities: int, method: str = 'euclidean') -> Tuple[dict, dict, dict]:
+def make_graph(n_sources: int, n_sinks: int, n_facilities: int, method: str = 'euclidean', random_seed: int = 12345) -> \
```
Comment (DKenefake) [inline src/energiapy/graph/graph.py:37]: changed method definition to include default seed, and changed return type sig to match function

File: src/energiapy/components/temporal_scale.py [hunk 3]
```diff
@@ -46,11 +46,11 @@ def __post_init__(self):
             name (str): the discretization list is the name.
             scale_levels (int): levels of the scale.
         """
-        self.scale = {i: [j for j in range(self.discretization_list[i])] for i in range(
-            len(self.discretization_list))}
-        self.list = [i for i in range(len(self.discretization_list))]
-        self.name = f"{[i for i in range(len(self.list))]}"
+
```
Comment (DKenefake) [inline src/energiapy/components/temporal_scale.py:49]: removed some unnecessary comprehensions

File: src/energiapy/graph/graph.py [hunk 4]
```diff
@@ -44,38 +44,44 @@ def make_graph(n_sources: int, n_sinks: int, n_facilities: int, method: str = 'e
...
         Tuple[dict, dict, dict]: _description_
     """
     source_list = list(range(n_sources))  # supply node
     sink_list = list(range(n_sinks))  # demand sink
     facility_list = list(range(n_facilities))
 
-    # electricity, gas, water
-    # criteria_utility_index = [i for i in source_list]
-    # critera_transport_index = [i for i in source_list]  # car, train
+    # random number generator with passed seed
```
Comment (DKenefake) [inline src/energiapy/graph/graph.py:55]: added a seedable prng generator and a helper function to make it less verbose

File: src/energiapy/model/constraints/cost.py [hunk 5]
```diff
@@ -63,19 +63,16 @@ def constraint_transport_imp_cost(instance: ConcreteModel, scheduling_scale_leve
     Returns:
         Constraint: transport_imp_cost
     """
-    scales = scale_list(instance=instance,
-                        scale_levels=scheduling_scale_level+1)
 
     def transport_imp_cost_rule(instance, sink, source, resource, transport, *scale_list):
         if trans_cost is None:
```
Comment (DKenefake) [inline src/energiapy/model/constraints/cost.py:68]: guard clause design pattern might make this more readable

File: src/energiapy/model/constraints/mode.py [hunk 6]
```diff
@@ -50,26 +55,34 @@ def constraint_production_mode_facility(instance: ConcreteModel, prod_max: dict,
...
+
+    if loc_pro_dict is None:
+        loc_pro_dict = dict()
+
     scales = scale_list(instance=instance,
-                        scale_levels=scheduling_scale_level+1)
+                        scale_levels=scheduling_scale_level + 1)
 
     def production_mode_facility_rule(instance, location, process, mode, *scale_list):
         if process in loc_pro_dict[location]:
```
Comment (DKenefake) [inline src/energiapy/model/constraints/mode.py:66]: gaurd pattern might make this more readable

File: src/energiapy/components/result.py [hunk 7]
```diff
@@ -81,7 +81,7 @@ def fetch_components(self, component_type: str, condition: tuple = None) -> set:
             return component_set
 
         else:
-            if (type(condition[1]) == bool or str) and (condition[1] not in ['eq', 'ge', 'le', 'g', 'l']):
+            if isinstance(condition[1], (bool, str)) and (condition[1] not in ['eq', 'ge', 'le', 'g', 'l']):
```
Comment (DKenefake) [inline src/energiapy/components/result.py:84]: probably a better way of doing this

**Comment by DKenefake** [comment]
Removed linting on markdown files, as I honestly don't see the need

File: src/energiapy/model/constraints/failure.py [hunk 8]
```diff
@@ -16,18 +16,23 @@
 from ...utils.scale_utils import scale_list
 
 
-def constraint_nameplate_production_failure(instance: ConcreteModel, fail_factor: dict = {}, network_scale_level: int = 0, scheduling_scale_level: int = 0) -> Constraint:
+def constraint_nameplate_production_failure(instance: ConcreteModel, fail_factor: dict = None, network_scale_level: int = 0, scheduling_scale_level: int = 0) -> Constraint:
```
Comment (DKenefake) [inline src/energiapy/model/constraints/failure.py:19]: changed default mutable argument to None, then a non  guard at the beginning of the function

File: src/energiapy/model/constraints/production.py [hunk 9]
```diff
@@ -27,26 +28,29 @@ def constraint_production_facility(instance: ConcreteModel, prod_max: dict, loc_
     Returns:
         Constraint: production_facility
     """
-    scales = scale_list(instance=instance, scale_levels=network_scale_level+1)
+    scales = scale_list(instance=instance, scale_levels=network_scale_level + 1)
 
     def production_facility_rule(instance, location, process, *scale_list):
         if loc_pro_dict is not None:
```
Comment (DKenefake) [inline src/energiapy/model/constraints/production.py:34]: guard pattern here as well

File: src/energiapy/components/result.py (hunk 7 above)
Comment (cacodcar) [reply src/energiapy/components/result.py:84]: Thanks :)

File: src/energiapy/components/temporal_scale.py (hunk 3 above)
Comment (cacodcar) [reply src/energiapy/components/temporal_scale.py:49]: no, you removed many.

File: src/energiapy/model/constraints/cost.py (hunk 5 above)
Comment (cacodcar) [reply src/energiapy/model/constraints/cost.py:68]: Gotcha

File: src/energiapy/model/constraints/cost.py [hunk 10]
```diff
@@ -63,19 +63,16 @@ def constraint_transport_imp_cost(instance: ConcreteModel, scheduling_scale_leve
...
-        elif distance_dict is None:
+
+        if distance_dict is None:
             return Constraint.Skip
-        else:
-            return instance.Trans_imp_cost[sink, source, resource, transport, scale_list[:scheduling_scale_level+1]] == trans_cost[transport]*distance_dict[(source, sink)]*instance.Trans_imp[sink, source, resource, transport, scale_list[:scheduling_scale_level+1]] 
-        
-        instance.constraint_transport_imp_cost = Constraint(instance.sinks, instance.sources, instance.resources_trans,
-                                                        instance.transports, *scales, rule=transport_imp_cost_rule, doc='import of resource from sink to source')
+
```
Comment (cacodcar) [inline src/energiapy/model/constraints/cost.py:73]: Jesus, I changed this yesterday and just forgot to put a return statement lol

File: src/energiapy/model/constraints/production.py (hunk 9 above)
Comment (cacodcar) [reply src/energiapy/model/constraints/production.py:34]: Gotcha

**Review by cacodcar — APPROVED** [review]
Thanks a ton man.
_Mog Asundi_

**Pull request merged**